src/components/stepindicator/StepIndicator/StepIndicator.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import classnames from 'classnames'
+import { StepIndicatorStepProps } from '../StepIndicatorStep/StepIndicatorStep'
+
+interface StepIndicatorProps {
+  showLabels?: boolean
+  counters?: true | false | 'small'
+  centered?: boolean
+  children: React.ReactElement<StepIndicatorStepProps>[]
+  className?: string
+  divProps?: JSX.IntrinsicElements['div']
+  listProps?: JSX.IntrinsicElements['ol']
+}
+export const StepIndicator = (
+  props: StepIndicatorProps
+): React.ReactElement => {
+  const {
+    showLabels = true,
+    counters = false,
+    centered = false,
+    children,
+    className,
+    divProps,
+    listProps,
+  } = props
+
+  const classes = classnames(
+    'usa-step-indicator',
+    {
+      'usa-step-indicator--no-labels': !showLabels,
+      'usa-step-indicator--counters': counters === true,
+      'usa-step-indicator--counters-sm': counters === 'small',
+      'usa-step-indicator--center': centered,
+    },
+    className
+  )
+
+  const findCurrentStepIndex = (): number => {
+    const i = children.findIndex((step) => step.props.status === 'current')
+    return i === -1 ? 0 : i
+  }
+  const currentStepIndex = findCurrentStepIndex()
+  const currentStepNumber = currentStepIndex + 1
+  const currentStepLabel = children[parseInt(`${currentStepIndex}`)].props.label
+  const totalNumberOfSteps = children.length
+
+  return (
+    <div className={classes} data-testid="step-indicator" {...divProps}>
+      <ol className="usa-step-indicator__segments" {...listProps}>
+        {children}
+      </ol>
+      <div className="usa-step-indicator__header">
+        <h2 className="usa-step-indicator__heading">
+          <span className="usa-step-indicator__heading-counter">
+            <span className="usa-sr-only">Step</span>
+            <span className="usa-step-indicator__current-step">
+              {currentStepNumber}
+            </span>
+            &nbsp;
```

Comment:
If you noticed this `&nbsp;` being present here but not in the [source code](https://designsystem.digital.gov/components/step-indicator/), I can add soem context:

In the examples, there are spaces as shown in the image below:
![image](https://user-images.githubusercontent.com/15805554/108868920-aeb3f080-75c4-11eb-9af7-7db99498af29.png)

These come from [implicit spacing added between `html` `inline` elements](https://stackoverflow.com/questions/5078239/how-do-i-remove-the-space-between-inline-inline-block-elements). 

Since rendered react html is minified, the implicit spacing is not rendered, so I had to explicitly add spaces here.

I may raise an issue with `uswds`. While this is not a quirky bug with a single browser, they may want to know about an inconsistency in display of one of their components when code is minified vs non-minified, similar to https://github.com/uswds/uswds/issues/3997.